src/core/thread/mle_router.cpp

```diff
@@ -3334,7 +3334,7 @@ ThreadError MleRouter::RestoreChildren(void)
     for (uint8_t i = 0; ; i++)
     {
         Child *child;
-        otChildInfo childInfo;
+        ChildInfo childInfo;
         uint16_t length;
 
         length = sizeof(childInfo);
```

Comment:
It is already handled by [otPlatSettingsGet()](https://github.com/openthread/openthread/blob/master/examples/platforms/utils/settings.cpp#L373), it limits the max read length which should not exceed the settings size in current software version. So current code could support backward compatibility naturally. Do you agree?